contracts/src/BoundlessMarket.sol

```diff
@@ -224,78 +225,113 @@ contract BoundlessMarket is
         // NOTE: Signature checks and recursive verification happen inside the assessor.
         bytes32[] memory requestDigests = new bytes32[](1);
         requestDigests[0] = fill.requestDigest;
-        bytes32 assessorJournalDigest =
-            sha256(abi.encode(AssessorJournal({requestDigests: requestDigests, root: claimDigest, prover: prover})));
+        bytes32 assessorJournalDigest = sha256(
+            abi.encode(
+                AssessorJournal({
+                    requestDigests: requestDigests,
+                    selectors: assessorFill.selectors,
+                    root: claimDigest,
+                    prover: assessorFill.prover
+                })
+            )
+        );
+        // Verification that the provided seal matches the required selector.
+        if (assessorFill.selectors.length == 1 && assessorFill.selectors[0].value != bytes4(fill.seal[0:4])) {
```

Comment:
There is a weird case here in that if `assessorFill.selectors.length > 1`, this check is not applied.